File: VB/WindowsFormsSample/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraPivotGrid
Imports System.IO
Imports DevExpress.Utils
Imports System.Collections
Imports DevExpress.XtraPivotGrid.Data
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Filtering
Imports DevExpress.Skins
Imports DevExpress.Utils.Drawing
Imports System.Diagnostics
Imports DevExpress.XtraPivotGrid.ViewInfo
Imports DevExpress.Utils.Serializing
Imports DevExpress.Utils.Serializing.Helpers

Namespace WindowsFormsSample
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			pivotGridControl1.Objects.Add(New Class1() With {.String = "test1"})
			pivotGridControl1.Objects.Add(New Class1() With {.String = "test2"})
			pivotGridControl1.Objects.Add(New Class1() With {.String = "test3"})
			pivotGridControl1.Object.String = "test4"
			pivotGridControl1.SaveLayoutToXml("test.xml")
			pivotGridControl1.Objects.Clear()
			pivotGridControl1.Object.String = Nothing
			pivotGridControl1.RestoreLayoutFromXml("test.xml")
		End Sub
	End Class
	Public Class MyPivotGrid
		Inherits PivotGridControl
		Public ReadOnly Property Objects() As List(Of Class1)
			Get
				Return Data.Objects
			End Get
		End Property
		Public ReadOnly Property [Object]() As Class1
			Get
				Return Data.Object
			End Get
		End Property
		Protected Overrides Function CreateData() As PivotGridViewInfoData
			Return New MyPivotGridViewInfoData(Me)
		End Function
		Public Shadows ReadOnly Property Data() As MyPivotGridViewInfoData
			Get
				Return TryCast(MyBase.Data, MyPivotGridViewInfoData)
			End Get
		End Property
	End Class
	Public Class MyPivotGridViewInfoData
		Inherits PivotGridViewInfoData
		Implements IXtraSupportDeserializeCollectionItem
		Public Sub New(ByVal control As IViewInfoControl)
			MyBase.New(control)
			_Objects = New List(Of Class1)()
		End Sub
		Private _Objects As List(Of Class1)
		<XtraSerializableProperty(XtraSerializationVisibility.Collection, True)> _
		Public ReadOnly Property Objects() As List(Of Class1)
			Get
				Return _Objects
			End Get
		End Property
		Private _Object As New Class1()
		<XtraSerializableProperty(XtraSerializationVisibility.Content)> _
		Public ReadOnly Property [Object]() As Class1
			Get
				Return _Object
			End Get
		End Property

		#Region "IXtraSupportDeserializeCollectionItem Members"

		Private Function CreateCollectionItem(ByVal propertyName As String, ByVal e As XtraItemEventArgs) As Object Implements IXtraSupportDeserializeCollectionItem.CreateCollectionItem
			If propertyName = "Objects" Then
				Dim class1 As New Class1()
				Objects.Add(class1)
				Return class1
			Else
				Return Nothing
			End If
		End Function

		Private Sub SetIndexCollectionItem(ByVal propertyName As String, ByVal e As XtraSetItemIndexEventArgs) Implements IXtraSupportDeserializeCollectionItem.SetIndexCollectionItem

		End Sub

		#End Region
	End Class
	Public Class Class1
		Private _String As String
		<XtraSerializableProperty> _
		Public Property [String]() As [String]
			Get
				Return _String
			End Get
			Set(ByVal value As String)
				_String = value
			End Set
		End Property
	End Class
End Namespace

File: CS/WindowsFormsSample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraPivotGrid;
using System.IO;
using DevExpress.Utils;
using System.Collections;
using DevExpress.XtraPivotGrid.Data;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Filtering;
using DevExpress.Skins;
using DevExpress.Utils.Drawing;
using System.Diagnostics;
using DevExpress.XtraPivotGrid.ViewInfo;
using DevExpress.Utils.Serializing;
using DevExpress.Utils.Serializing.Helpers;

namespace WindowsFormsSample {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
        private void Form1_Load(object sender, EventArgs e) {
            pivotGridControl1.Objects.Add(new Class1() { String = "test1" });
            pivotGridControl1.Objects.Add(new Class1() { String = "test2" });
            pivotGridControl1.Objects.Add(new Class1() { String = "test3" });
            pivotGridControl1.Object.String = "test4";
            pivotGridControl1.SaveLayoutToXml("test.xml");
            pivotGridControl1.Objects.Clear();
            pivotGridControl1.Object.String = null;
            pivotGridControl1.RestoreLayoutFromXml("test.xml");   
        }
    }
    public class MyPivotGrid : PivotGridControl {
        public List<Class1> Objects {
            get {
                return Data.Objects;
            }
        }
        public Class1 Object {
            get { return Data.Object; }
        }
        protected override PivotGridViewInfoData CreateData() {
            return new MyPivotGridViewInfoData(this);
        }
        public new MyPivotGridViewInfoData Data {
            get {
                return base.Data as MyPivotGridViewInfoData;
            }
        }
    }
    public class MyPivotGridViewInfoData : PivotGridViewInfoData, IXtraSupportDeserializeCollectionItem {
        public MyPivotGridViewInfoData(IViewInfoControl control) : base(control) {
            _Objects = new List<Class1>();
        }
        private List<Class1> _Objects;
        [XtraSerializableProperty(XtraSerializationVisibility.Collection, true)]
        public List<Class1> Objects {
            get {
                return _Objects;
            }
        }
        private Class1 _Object = new Class1();
        [XtraSerializableProperty(XtraSerializationVisibility.Content)]
        public Class1 Object {
            get { return _Object; }
        }
        
        #region IXtraSupportDeserializeCollectionItem Members

        object IXtraSupportDeserializeCollectionItem.CreateCollectionItem(string propertyName, XtraItemEventArgs e) {
            if (propertyName == "Objects") {
                Class1 class1 = new Class1();
                Objects.Add(class1);
                return class1;
            } else return null;
        }

        void IXtraSupportDeserializeCollectionItem.SetIndexCollectionItem(string propertyName, XtraSetItemIndexEventArgs e) {
            
        }

        #endregion
    }
    public class Class1 {
        private string _String;
        [XtraSerializableProperty]
        public string String {
            get { return _String; }
            set {
                _String = value;
            }
        }
    }
}
